lib/hcaptcha_verifier.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module HcaptchaVerifier
+  include SemanticLogger::Loggable
+
+  class << self
+    VERIFY_ENDPOINT = "https://hcaptcha.com/siteverify"
+    SIGN_IN_CAPTCHA_THRESHOLD = 4
+    SIGN_UP_CAPTCHA_THRESHOLD = 2
+    TIMEOUT_SEC = 5
+
+    def should_verify_sign_in?(user_display_id)
+      key = Rack::Attack::LOGIN_THROTTLE_PER_USER_KEY
+      period = Rack::Attack::LOGIN_LIMIT_PERIOD
+
+      keys = keys(key, period, user_display_id)
+      should_verify(keys, SIGN_IN_CAPTCHA_THRESHOLD)
+    end
+
+    def should_verify_sign_up?(ip)
+      key = Rack::Attack::SIGN_UP_THROTTLE_PER_IP_KEY
+      period = Rack::Attack::SIGN_UP_LIMIT_PERIOD
+
+      keys = keys(key, period, ip)
+      should_verify(keys, SIGN_UP_CAPTCHA_THRESHOLD)
+    end
+
+    def call(client_response, remote_ip)
```

Comment:
_nitpick: I'm personally not big fan of `call` methods if not used as a proc. Would it make sense to rename this to something like `verified?`. Similar change could be done in PrivacyPassRedeemer._